src/datamodel/validator/Uuid.cpp

```diff
@@ -10,3 +10,139 @@
+	}
+}
+
+using ValidationResult = std::tuple<bool, std::optional<Reason>>;
+
+ValidationResult isUuid(const uprotocol::v1::UUID uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t lsb = uuid.lsb();
+
+	uint8_t version = (msb >> 12) & 0xF;
+	if (version != 8) {
+		return {false, Reason::WRONG_VERSION};
+	}
+
+	uint8_t variant = (lsb >> 62) & 0x3;
+	if (variant != 2) {  // Variant should be 10 in binary
+		return {false, Reason::UNSUPPORTED_VARIANT};
+	}
+
+	uint64_t timestamp =
+	    (msb >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
+	uint64_t current_time =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	if (timestamp > current_time) {
+		return {false, Reason::FROM_THE_FUTURE};
+	}
+
+	return {true, std::nullopt};
+}
+
+ValidationResult isExpired(const uprotocol::v1::UUID uuid,
+                           std::chrono::milliseconds ttl) {
+	auto [valid, maybe_reason] = isUuid(uuid);
+	if (!valid) {
+		return {false, maybe_reason};
+	}
+
+	uint64_t timestamp =
+	    (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
+	uint64_t current_time =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	if ((current_time - timestamp) > ttl.count()) {
+		return {false, Reason::EXPIRED};
+	}
+
+	return {true, std::nullopt};
+}
+
+uint8_t getVersion(const uprotocol::v1::UUID uuid) {
+	auto [valid, maybe_reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid("Invalid UUID version");
+	}
+	return (uuid.msb() >> 12) & 0xF;
+}
+
+uint8_t getVariant(const uprotocol::v1::UUID uuid) {
+	auto [valid, maybe_reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid("Invalid UUID variant");
+	}
+	return (uuid.lsb() >> 62) & 0x3;
+}
+
+std::chrono::system_clock::time_point getTime(const uprotocol::v1::UUID uuid) {
+	auto [valid, maybe_reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid("Invalid UUID timestamp");
+	}
+	uint64_t timestamp = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;
+	return std::chrono::system_clock::time_point(
+	    std::chrono::milliseconds(timestamp));
+}
+
+std::chrono::milliseconds getElapsedTime(const uprotocol::v1::UUID uuid) {
+	auto [valid, maybe_reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid("Invalid UUID timestamp");
+	}
+
+	uint64_t current_time =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	uint64_t uuid_time = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;
+	return std::chrono::milliseconds(current_time - uuid_time);
+}
+
+std::chrono::milliseconds getRemainingTime(const uprotocol::v1::UUID uuid,
+                                           std::chrono::milliseconds ttl) {
+	auto elapsed_time = getElapsedTime(uuid);
+	if (elapsed_time > ttl) {
+		return std::chrono::milliseconds(0);
+	}
+	return ttl - elapsed_time;
```

Comment:
Minor: This could be condensed
```suggestion
	auto elapsed_time = getElapsedTime(uuid);
	return std::min(ttl - elapsed_time, 0ms);
```